### ISSUE TITLE
Little "Could Be a bug" Bug Report

### ISSUE THREAD

**Issue by Speiger** [WanionCane/UniDict#3]
I have found randomly that mod and find it interesting...
After a little checking that you support IC2.
Now here is the thing (at least for 1.7.10 because 1.9.4/1.10 is still in progress),
did you take IC2 Classic into account?
While the Machine Recipe Managers should be fine the Crafting Recipes could cause issues, because of differend class paths & field names & differned way of handling stuff...
Could you take care of that too? (If you still support 1.7.10?) (And later for 1.9.4/1.10?)

**Comment by WanionCane** [comment]
i thought adding support for ic2 classic, but they share the same modid, so, if there is a way to detect ic2 classic, i would like to know.

**Comment by WanionCane** [comment]
and yes, i will be pleased to take care of this, personally, i prefer the classic ic2, reminds me of the old times.

**Comment by WanionCane** [comment]
sorry for all these messages, i'm kind new to github and such, lol
i'm still supporting 1.7.10, but i giving a time to the changes that i did in 1.10.2, to check stability and another things, since are really a lot of changes, i'm planing just do backport it to 1.7.10, will be way easier.
i also want to ask you something, i "ask" something in my curseforge page, and nobody answered it yet.
would be nice to have your opnion:
[Link to the post](http://minecraft.curseforge.com/projects/unidict?comment=85)

> hi guys.
> i'm planning to backport UniDict for three old minecraft version.
> 1.4.6, 1.5.2, 1.6.4.
> since there are no new mod for these versions, i will have to do it only once.
> i want to hear from you guys about this, also, help me, by suggesting (mod)integration for these versions.
> thank you

everyone that i asked about this said that would be a waste of time, but i don't know, i'm really feeling that it worth.

**Comment by Speiger** [comment]
@WanionCane well first thanks.
The second ModID is: "IC2-Classic-Spmod"
Also no problem with multimessages^^" only thing is i was already asleep.

Here by the way is the RecipeHelper class for IC2 Classic. When this mod is loaded you just needs its API to detect & access recipes. (Note: the getRecipeList function accesses not the CraftingManager. It accesses a custom Cache which is unmodifyable... (For Restoring Recipes or keep track which recipes got added)
https://github.com/TinyModularThings/IC2Classic/blob/1.7.10/src/main/java/ic2/api/recipe/ICraftingRecipeManagerList.java
This class is only for 1.7.10 i have already a 1.9.4/1.10 version of it but its not finished because its missing a couple features... (AdvRecipe & AdvShapeless recipe are also not implemented in 1.9.4 yet because its the last thing i implement)

To your question now: Well i do not think that is a waste, the only thing is you properly will not get downloads for that or bugreports or rarly someone will use it... But that will be not your main problem...
The main problem you will have is making a WorkSpace for the MC versions because the MC Launcher did change & file storage too... The problem there is that you do not get all needed files together...
(Else i would support 1.6.4 if someone asks for it, (Only bugs))

**Comment by WanionCane** [comment]
Thank you very much for the help, as soon i start messing with 1.7.10 I will include IC2 Classic.
also, can you take a look at this class to see if it is compatible with ic2 classic?
[wanion.unidict.recipe.IC2Researcher.java](https://github.com/WanionCane/UniDict/blob/1.10.2/src/main/java/wanion/unidict/recipe/IC2Researcher.java)
UniDict uses it to "figure out" the shape of recipes.

**Comment by Speiger** [comment]
^^"
I can tell you it will crash. Because my Classes are in a differend folder...
Now i have to note: IC2C has not bin released yet for 1.9 so you are fine.
As soon we go into testing phase i will upload the API tell you when it is ready...
Also to mention: IC2 May support the RecipeAPI of Exp but will mainly drop it and do its own thing with recipes, because what player is doing with recipes is very limited & modpacks have no real freedome...
So i decided to drop it. (This count for 1.9/1.10) (It is still supported for those who want only to support the Exp API (simply to keep compat) but there is a more accessable hook into it...
Edit: This counts for crafting & Machine Recipes.

**Comment by WanionCane** [comment]
I am working on it, i will keep you updated, ok? ^^
may take awhile, because all the changes that i did in 1.10.2

**Comment by Speiger** [comment]
I am thinking that i need 2-4 Weeks to get the first test version ready... (Hoping)
And yeah that would be great :3

**Comment by WanionCane** [comment]
oh, i discovered something interesting, the current 1.7.10 IC2 Integration is fully compatible with IC2 Classic.

**Comment by Speiger** [comment]
Yeah because of 1 thing. You are not using the Crafting Recipes and that was the only thing that used IC2 Core classes... So yeah thats good.
Oh by the way there is 1 feature comming with IC2C and that is stacksize crafting. (Example: You can say 5 ironIngots + 1 Gold Dust = Mixed ingot (Using 2 Slots)) So that you should support then too

**Comment by WanionCane** [comment]
you mean, multiple outputs? if yes, it is already supported because of the Thermal Centrifuge.
since IC2 Classic is already supported with no issues, may I include it in the integration list?

**Comment by Speiger** [comment]
xD No. Crafting Table Inputs... (So with 1.9 you can actually make more then 1 Item per slot (stacksize) as requirement xD

**Comment by WanionCane** [comment]
=O? are you serious? omg, this is awesome, thanks for letting me know =D

**Comment by Speiger** [comment]
Its not as easy as you think. but the function:
ItemStack[] getRemainingItems(InventoryCrafting inv);
is allowing you to do it because you can actually override what is in the crafting menu after crafting...

**Comment by WanionCane** [comment]
since i can know the recipe ingredients, i will just call for recipe.matches(InventoryCrafting inv) with 64 as stacksize, the difference will show up.
but i believe to achieve this, i will have to do a custom recipe class.

**Comment by Speiger** [comment]
Its getRemain Items which modify the items. Matches tells you only if the recipe can be performed

**Comment by WanionCane** [comment]
ok, i will take a deep look in it later, 
i released [UniDict-1.7.10-2.8.jar](http://minecraft.curseforge.com/projects/unidict/files/2317733) on curse.
it is fully compatible with both IC2 Experimental and IC2 Classic.
even with the recipes. as you can see here: [wanion.unidict.recipe.IC2RecipeResearcher](https://github.com/WanionCane/UniDict/blob/1.7.10/src/main/java/wanion/unidict/recipe/IC2RecipeResearcher.java).
tell me if i missed something, also, thank you very much for your help ^^

as soon you release IC2Classic for 1.9.4+ i will make all the changes if are required to it be compatible with both experimental and classic.

**Comment by Speiger** [comment]
Cool :+1:  xD
I will probably upload the api even before the release (that will be all you need) and when i have a testversion i can give you (not public) the test version xD

**Comment by WanionCane** [comment]
yeah, i would be glad to receive it first XD

**Comment by WanionCane** [comment]
any news? :O

**Comment by Speiger** [comment]
@WanionCane well just finishing the reactorplanner... Then there i still have a lot of things todo...
Specially with the API there will join more RecipeChanges...
^^" And i switched to 1.10.2.
Here is my todolist: (Reactorplanner was the last big thing on the list)
https://github.com/TinyModularThings/IC2Classic/milestone/4

**Comment by WanionCane** [comment]
really cool =D
I also saw this, [IC2 Classic Update News](https://www.reddit.com/r/feedthebeast/comments/53nldl/ic2_classic_update_news/)
it is... amazing XD

**Comment by Speiger** [comment]
Lol^^ but I feel that people wish that I was already finished

**Comment by Speiger** [comment]
well we are now in the beta so just to let you know^^"

**Comment by WanionCane** [comment]
really cool =p

**Comment by Speiger** [comment]
API by the way is public so you can read the documentation of it.
https://github.com/TinyModularThings/IC2Classic/tree/master/src/main/java/ic2/api
The classic folder is basicly anything classic only related.

**Comment by WanionCane** [comment]
Thanks, soon I will take a look =p
